src/main/java/gg/solarmc/kitpvp/messaging/Message.java

```diff
@@ -0,0 +1,37 @@
+package gg.solarmc.kitpvp.messaging;
+
+import org.bukkit.entity.Player;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class Message {
```

Comment:
This class and its compatriots are an interesting way of doing placeholders I didn't think of before. I've always parsed placeholders manually based on the context. Some things I'm wondering:

* Why use a List<String> and not a String including line-breaks? The latter would be more easily operated upon
* Have you considered using adventure rather than ChatColor.translateAlternateColorCodes? If we code against Components in our plugins now, it will be easy to add support for fancier text like hover events later on.

Reply:
This class exists so that i don't have to define a %fucking_placeholder% for every message that exists, relative to context
I'll definetly switch it around to support adventure

Reply:
Yes, I think it's a good way of doing placeholders